### PR TITLE
Added a pre-commit hooks for developer use

### PR DESCRIPTION
**Pull request by zhangcandrew**
Automatically updates version.txt when committing if scripts staged for
changes have not been updated within the script or within version.txt.

Hook acts before a commit.

### CODE REVIEW

**Comment by zhangcandrew** [comment]
@henrykironde I switched over to this new PR. 

in response to: 
```
else:
# os.system("ln -s -f ../../hooks/pre-commit .git/hooks/pre-commit")
```

I was using it just in case the developer already had git hooks in place. We create a symbolic link from our hook to theirs instead of overwriting their file so that they can keep their own git hooks along with the ones we have for them. I'm not sure if it's too useful, but I added it as a precaution.

File: setup.py [hunk 1]
```diff
@@ -4,10 +4,35 @@
...
 
 current_platform = platform.system().lower()
 extra_includes = []
+if current_platform == "darwin":
+    if(subprocess.check_output("[ ! -f .git/hooks/pre-commit ] && echo \"Not Found\"") == "Not Found"):
+      os.system("chmod +x hooks/pre-commit")
+      os.system("cp -p hooks/pre-commit .git/hooks/pre-commit")
+    else:
+      os.system("ln -s -f ../../hooks/pre-commit .git/hooks/pre-commit")
+    try:
```
Comment (henrykironde) [inline setup.py]:
```
    try:
         import py2app
     except ImportError:
         pass
     extra_includes = []
```

not needed

File: setup.py [hunk 2]
```diff
@@ -4,10 +4,35 @@
...
+        import py2app
+    except ImportError:
+        pass
+    extra_includes = []
+elif current_platform == "windows":
+    os.system("chmod +x hooks/pre-commit")
+    os.system("mklink hooks/pre-commit .git/hooks/pre-commit")
+    try:
```
Comment (henrykironde) [inline setup.py]:
```
     try:
         import py2exe
     except ImportError:
         pass
     import sys
     extra_includes = ['pyodbc', 'inspect']
     sys.path.append(
         "C:\\Windows\\winsxs\\x86_microsoft.vc90.crt_1fc8b3b9a1e18e3b_9.0.21022.8_none_bcb86ed6ac711f91")
```
not used any more

**Comment by henrykironde** [comment]
I think with those changes we shall have be ready to test :+1:

File: setup.py [hunk 3]
```diff
@@ -4,10 +4,23 @@
...
 from pkg_resources import parse_version
 import platform
+import sys
+import warnings
 import os
+import subprocess
 
 current_platform = platform.system().lower()
 extra_includes = []
+if current_platform == "darwin":
```
Comment (henrykironde) [inline setup.py]: If Linux and OS x have the same procedures 
we can say 
```
if window:
    do all the windows checks and commands
else : 
     do command for both mac os x and linux
```

**Comment by henrykironde** [comment]
let me know what you think

File: setup.py (hunk 3 above)
Comment (zhangcandrew) [inline setup.py]: If OS X and Linux have the same procedures, that would work. I have to test it on Linux still, so I will circle back to you on that!

Comment (zhangcandrew) [inline setup.py]: Yeah, it seems to be working on Linux now. Another problem we now face though is that the commands module does not run in python 3.5. I might need to switch the entire pre-commit over to subprocess

Comment (henrykironde) [inline setup.py]: go for it

File: setup.py [hunk 4]
```diff
@@ -4,10 +4,23 @@
...
+    if(subprocess.check_output("[ ! -f .git/hooks/pre-commit ] && echo \"Not Found\"") == "Not Found"):
+      os.system("chmod +x hooks/pre-commit")
+      os.system("cp -p hooks/pre-commit .git/hooks/pre-commit")
+    else:
+      os.system("ln -s -f ../../hooks/pre-commit .git/hooks/pre-commit")
+
+elif current_platform == "windows":
```
Comment (zhangcandrew) [inline setup.py]: I think the mklink serves the same purpose in setup.py, but I could be wrong.

**Comment by zhangcandrew** [comment]
Just updated pre-commit and setup.py. I'm currently using the git hook, and I have tested it with a few scripts. Everything seems to be ok for now on Ubuntu. Let me know how things go on your side!

**Comment by henrykironde** [comment]
Closing because I have added the changes to a new pr

**Pull request closed without merging**